### PR TITLE
[improvement-merging-maps] merge maps into one

### PR DESCRIPTION
**Pull request by sk1418**

### CODE REVIEW

File: core-java-modules/core-java-streams-simple/src/test/java/com/baeldung/streams/flatmap/map/Java8MapAndFlatMapUnitTest.java [hunk 1]
```diff
@@ -48,4 +49,35 @@ void givenOptional_whenCalledFlatMap_thenProduceFlattenedOptional() {
             .flatMap(s -> Optional.of("STRING")));
     }
 
-}
+    @Test
+    void givenMaps_whenCalledFlatMap_thenMultipleMapsMergedIntoOneMap() {
```
Comment (theangrydev) [inline core-java-modules/core-java-streams-simple/src/test/java/com/baeldung/streams/flatmap/map/Java8MapAndFlatMapUnitTest.java:53]: Can we add one more test to show the limitation of this approach when there are duplicate keys, if I remember correctly Collectors.toMap will throw an exception on duplicate keys

Comment (theangrydev) [reply core-java-modules/core-java-streams-simple/src/test/java/com/baeldung/streams/flatmap/map/Java8MapAndFlatMapUnitTest.java:53]: Similar for null keys I think
